MaxText/train.py

```diff
@@ -722,19 +907,23 @@ def train_loop(config, state=None):
         cumulative_eval_metrics["scalar"]["eval/total_loss"] += float(eval_metrics["scalar"]["evaluation/total_loss"])
         cumulative_eval_metrics["scalar"]["eval/total_weights"] += float(eval_metrics["scalar"]["evaluation/total_weights"])
         cumulative_eval_metrics["scalar"]["eval/moe_lb_loss"] += float(eval_metrics["scalar"]["evaluation/moe_lb_loss"])
-        max_logging.log(f"Completed eval step {eval_step_count}")
```

Comment:
Can we keep this line? tfds and grain pipeline may hang when eval_steps were not set correctly and eval data run out, this line help users track how many eval steps can be completed.

Reply:
oh, nvm, I see it's moved to later

Reply:
But it was out of order with the iterator increment, so it needed a fix, thanks for catching this!